### PR TITLE
Type hints and decorator for TF T5

### PR DESCRIPTION
**Pull request by Dahlbomii**
@Rocketknight1 @gante

### CODE REVIEW

**Comment by Rocketknight1** [comment]
Thanks for this, the PR looks perfect to me! I see XLA generation tests failing, though - @gante could this be another situation like GPT-2 with some weird input names that are exposed by the decorator?

File: src/transformers/models/t5/modeling_tf_t5.py [hunk 1]
```diff
@@ -181,7 +181,6 @@ def __init__(self, config, has_relative_attention_bias=False, **kwargs):
         self.output_attentions = config.output_attentions
 
         self.relative_attention_num_buckets = config.relative_attention_num_buckets
-        self.relative_attention_max_distance = config.relative_attention_max_distance
```
Comment (gante) [inline src/transformers/models/t5/modeling_tf_t5.py:184]: Is there a particular reason why this was removed? 🤔

File: src/transformers/models/t5/modeling_tf_t5.py [hunk 2]
```diff
@@ -1547,7 +1473,6 @@ def prepare_inputs_for_generation(
         input_ids,
         past=None,
         attention_mask=None,
-        decoder_attention_mask=None,
```
Comment (gante) [inline src/transformers/models/t5/modeling_tf_t5.py:1550]: I don't think this should be removed as well

File: src/transformers/models/t5/modeling_tf_t5.py [hunk 3]
```diff
@@ -1565,7 +1490,6 @@ def prepare_inputs_for_generation(
             "past_key_values": past,
             "encoder_outputs": encoder_outputs,
             "attention_mask": attention_mask,
-            "decoder_attention_mask": decoder_attention_mask,
```
Comment (gante) [inline src/transformers/models/t5/modeling_tf_t5.py:1568]: (ditto)

File: src/transformers/models/t5/modeling_tf_t5.py [hunk 4]
```diff
@@ -288,7 +287,6 @@ def compute_bias(self, query_length, key_length):
             relative_position,
             bidirectional=(not self.is_decoder),
             num_buckets=self.relative_attention_num_buckets,
-            max_distance=self.relative_attention_max_distance,
```
Comment (gante) [inline src/transformers/models/t5/modeling_tf_t5.py:291]: (ditto)

**Review by gante — COMMENTED** [review]
Almost good -- there were a few lines that got removed (and they shouldn't)

**Comment by Rocketknight1** [comment]
@gante I think some of those were changes to main that came in during the PR, including the missing XLA function that caused those test failures! Let me fix it up real quick.

**Review by gante — APPROVED** [review]

**Comment by Rocketknight1** [comment]
@Dahlbomii Tests are green so we're going to merge now. Thanks for your help with this, and thanks for your patience as we both edited the file you were working on and renamed `master` to `main` midway through your PR!

**Pull request merged**